src/legacy/core_plugins/data/public/expression_executor/expression_executor_service.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup } from 'kibana/public';
+import { useRef, useEffect } from 'react';
+import React from 'react';
```

Comment:
react shouldn't be part of the API, we can have react components wrapping the api but pls put in separate file

Reply:
So we don't do stateful components at all? Or is this just about the React import in the service itself?

The nice thing about delivering components like that is that they are usable directly (`<ExpressionRenderer expression="..." />`) instead of having to pass the stateful thing in (e.g. `<ExpressionRenderer runFn={plugins.expressionExecutor.run} expression="..." />`)

Reply:
Yeah agreed with @ppisljar here -- I'd suggest providing a `run` function to just run the expression, and a in a separate file a "convenience" method for React, which uses `run` under the hood.

Then both can be returned from `setup` as part of the public contract. This way people have the option of using the React wrapper or the `run` method directly. Also some people might want to run an expression and get the output without having to use a renderer.